src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/CollectionPropertiesShouldBeReadOnly.cs

```diff
@@ -57,11 +57,11 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.RegisterCompilationStartAction(
                 (context) =>
                 {
-                    INamedTypeSymbol iCollectionType = WellKnownTypes.ICollection(context.Compilation);
-                    INamedTypeSymbol genericICollectionType = WellKnownTypes.GenericICollection(context.Compilation);
-                    INamedTypeSymbol arrayType = WellKnownTypes.Array(context.Compilation);
-                    INamedTypeSymbol dataMemberAttribute = WellKnownTypes.DataMemberAttribute(context.Compilation);
-                    ImmutableHashSet<INamedTypeSymbol> immutableInterfaces = WellKnownTypes.IImmutableInterfaces(context.Compilation);
+                    INamedTypeSymbol iCollectionType = context.Compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemCollectionsICollection);
+                    INamedTypeSymbol genericICollectionType = context.Compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemCollectionsGenericICollection1);
+                    INamedTypeSymbol arrayType = WellKnownTypeProvider.GetOrCreate(context.Compilation).SystemArray;
```

Comment:
This seems to have introduced inconsistency now. It is unclear why you would use `compilation.GetTypeByMetadataName` for one well known type and `WellKnownTypeProvider.GetOrCreate` for another. Previous implementation was much more readable and consistent.